File: dotnet/HotelBooking/HotelBooking/Models/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace HotelBooking.Models;

public class Booking
{
    public int Id { get; set; }
    
    public DateTime CheckIn { get; set; }
    
    public DateTime CheckOut { get; set; }
    
    [System.Text.Json.Serialization.JsonIgnore]
    public Customer? Customer { get; set; }
    
    public int CustomerId { get; set; }
    
    [System.Text.Json.Serialization.JsonIgnore]
    public Room? Room { get; set; }
    
    public int RoomId { get; set; }

    public int StayLength
    {
        get
        {
            var stayLength = (CheckOut - CheckIn).Days;
            return stayLength > 1 ? stayLength : 1;
        }
    }
    
    public int Bill => StayLength * Room?.RoomPrice ?? 0;
}

File: java/hotelbooking/src/main/java/no/egde/hotelbooking/models/Booking.java
package no.egde.hotelbooking.models;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private LocalDateTime checkIn;
    private LocalDateTime checkOut;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "customerId", referencedColumnName = "id")
    private Customer customer;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "roomId", referencedColumnName = "id")
    private Room room;

    private int stayLength;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public LocalDateTime getCheckIn() {
        return checkIn;
    }

    public void setCheckIn(LocalDateTime checkIn) {
        this.checkIn = checkIn;
    }

    public LocalDateTime getCheckOut() {
        return checkOut;
    }

    public void setCheckOut(LocalDateTime checkOut) {
        this.checkOut = checkOut;
    }

    public Customer getCustomer() {
        return customer;
    }

    public void setCustomer(Customer customer) {
        this.customer = customer;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public int getStayLength() {
        return stayLength;
    }

    public void setStayLength(int stayLength) {
        this.stayLength = stayLength;
    }

    public int getBill() {
        return getStayLength() * getRoom().getRoomPrice();
    }
}
